hudi-common/src/main/java/org/apache/hudi/io/storage/HoodieAvroFileReaderBase.java

```diff
@@ -37,7 +37,7 @@ abstract class HoodieAvroFileReaderBase implements HoodieAvroFileReader {
   @Override
   public ClosableIterator<HoodieRecord<IndexedRecord>> getRecordIterator(Schema readerSchema, Schema requestedSchema) throws IOException {
     ClosableIterator<IndexedRecord> iterator = getIndexedRecordIterator(readerSchema, requestedSchema);
-    return new MappingIterator<>(iterator, data -> unsafeCast(new HoodieAvroIndexedRecord(data)));
+    return new CloseableMappingIterator<>(iterator, data -> unsafeCast(new HoodieAvroIndexedRecord(data)));
```

Comment:
Can we move these changes not related to Bloom Index into a separate PR?

Reply:
I think i know how this crippled in. Let me clean it out